include/openthread/platform/crypto.h

```diff
@@ -37,15 +37,9 @@
 
 #include <stdint.h>
 
-#include "openthread-core-config.h"
-
-#include <mbedtls/aes.h>
-#include <mbedtls/md.h>
-#include <psa/crypto_types.h>
 #include <openthread/error.h>
 #include <openthread/platform/radio.h>
 #include "psa/crypto.h"
```

Comment:
Is this header still used in this file? I have not found any reference in this interface.

Reply:
@hubertmis Yes, this is needed and cannot be removed. This is because we use `psa_key_attributes_t` in `crypto_platform.cpp`. This is a structure and not a basic datatype. For this I have added the macro `OT_PLAT_CRYPTO_KEY_ATTRIBUTES_TYPE` to be defined by the platforms. If PSA is used this will be defined to `psa_key_attributes_t`. Without the crypto.h header, compilation fails.

Reply:
Looks like "psa/crypto.h" is a third party header. If so, we should use `#include <psa/crypto.h>` style for it.

Reply:
Understood. I will change this. @abtink how do we handle the API version increments? Is there any writeup to explain this?

Reply:
When we have any API change (platform or public), the `OPENTHREAD_API_VERSION` in `instance.h` should be incremented. There is a GitHub CI check that ensures that this is done on a PR if it changes any of the public/platform headers.

Reply:
> This is because we use `psa_key_attributes_t` in `crypto_platform.cpp`.

Are files up-to-date now? I can't find `psa_key_attributes_t` in this file or in `crypto_platform.cpp`.
Anyway, if this header is used in `crypto_platform.cpp` implementation, I think it should be included in this file. Not in generic interface like `platform/crypto.h`.

Reply:
Yes they are up-to-date. Right now the implementation is as follows:

`crypto.h` needs to define `otCryptoKeyAttributes`as this will be needed to the `otPlatCryptoGetKeyAttributes` uses this. In case of PSA, this should be typedef to `psa_key_attributes_t`. Without this header in `crypto.h` the compiler errors out. The only other solution is to introduce inclusion od spa header inside core files like `key_manager.cpp`

`otCryptoKeyAttributes` is defined as follows:

```/**
 * This is the default definition of OT_PLAT_CRYPTO_KEY_ATTRIBUTES_TYPE.
 *
 */

#ifndef OT_PLAT_CRYPTO_KEY_ATTRIBUTES_TYPE
#define OT_PLAT_CRYPTO_KEY_ATTRIBUTES_TYPE uint8_t
#endif

typedef OT_PLAT_CRYPTO_KEY_ATTRIBUTES_TYPE otCryptoKeyAttributes;```

The platform can provide its own definition of OT_PLAT_CRYPTO_KEY_ATTRIBUTES_TYPE to over-ride this definition. In case of PSA this would be `psa_key_attributes_t`.

Reply:
Oh, I see now. You are simply overriding `uint8_t` with other one with preprocessor macros. I'm afraid this solution is not scalable, because if you introduce a new crypto engine implementation, you would need to modify interface to support other key attributes type.

I think the best solution to keep implementation-specific data would be to define `otCryptoKeyAttributes` as an array like:
```
typedef uint8_t otCryptoKeyAttributes[OT_PLAT_CRYPTO_KEY_ATTRIBUTES_SIZE];
```
and override the size during building process, instead of the type itself.

Alternatively, you could replace `#include <psa/crypto.h>` with `#include OT_PLAT_CRYPTO_KEY_ATTRIBUTES_TYPE_HEADER` to override this header alongside the type.

Reply:
Understood. I agree the best solution would be to have an array. I will change the implementation.